test/tap/tests/mysql-protocol_compression_level-t.cpp

```diff
@@ -0,0 +1,184 @@
+#include <cstdlib>
+#include <cstdio>
+#include <cstring>
+#include <unistd.h>
+
+#include <string>
+#include <sstream>
+#include <fstream>
+#include "mysql.h"
+
+#include "tap.h"
+#include "command_line.h"
+#include "utils.h"
+
+inline unsigned long long monotonic_time() {
+  struct timespec ts;
+  clock_gettime(CLOCK_MONOTONIC, &ts);
+  return (((unsigned long long) ts.tv_sec) * 1000000) + (ts.tv_nsec / 1000);
+}
+
+unsigned long calculate_query_execution_time(MYSQL* mysql, const std::string& query) {
+	MYSQL_RES *res;
+	MYSQL_ROW row;
+	unsigned long long begin = monotonic_time();
+	unsigned long long row_count = 0;
+	MYSQL_QUERY(mysql, query.c_str());
```

Comment:
Using this macro here can have unintended consequences. When failing to perform a query, this macro will return `EXIT_FAILURE` (1), so, if for any reason both queries fail to be executed (broken connection for example), the test for checking the execution time:

```
unsigned long diff = abs(time_proxy - time_proxy_compressed);
int performance_diff = (diff * 100) / time_proxy;
```

Would still succeed, as it would be checking `1 - 1 = 0`, and `0` passes the `ok` check right now as a value. Error handling should be encoded here in the return value itself, as they collide with real values from the query/computation.

Reply:
Fixed. Now checking for error and returing -1 to indicate error.